mergify_engine/rules/__init__.py

```diff
@@ -459,7 +594,7 @@ def get_pull_request_rules_schema(partial_validation: bool = False) -> voluptuou
                     voluptuous.Required("name"): str,
                     voluptuous.Required("hidden", default=False): bool,
                     voluptuous.Required("conditions"): voluptuous.All(
-                        [voluptuous.All(str, voluptuous.Coerce(RuleCondition))],
+                        [voluptuous.Coerce(RuleConditionSchema)],
```

Comment:
This means that this is valid:
```
conditions:
  or:
    - foobar
    - foobaz
```

I think it's neat, but we might want to add a test?

Reply:
That looks good, but I will focus on the current structure and add this neat syntax later.